specs/01-sharded-archiving.md

```diff
@@ -0,0 +1,288 @@
+
+```
+| source_share_0 | π_0 | C_bs_0 |
+| source_share_1 | π_1 | Cb_1 |
+...
+| source_share_M | π_M | Cb_M |
+| parity_share_0 | π_s0 | Cbs_0 |
+| parity_share_1 | π_s1 | Cbs_1 |
+...
+| parity_share_M | π_sM | Cbs_M |
+```
+
+- When the proof for the block is generated, a new `BlockInfo` data structure is generated that is
+  used to submit the block information to the parent chain. Any node (not only farmers) are allowed
+  to perform this operation.
+- New encoded blocks need to be stored by farmers and made available for retrieval by other shards
+  and nodes in the system.
+
+```rust
+type BlockInfo {
+    shard_id: [u8; 20]
+    block_height: u64
+    block_hash: [u8; 32]
+    block_proof: [u8;32]
+}
+```
+
+**Table 1: Block Encoding Parameters**
+
+| Parameter              | Description                 | Example Value (Assumed) |
+| ---------------------- | --------------------------- | ----------------------- |
+| `block_size`           | Size of the block in bytes  | 4,000,000 bytes         |
+| `SHARE_SIZE`           | Size of each share in bytes | 4 kbytes                |
+| `source_share`         | Total number of shares      | 100                     |
+| `total_shares`         | Shares after erasure coding | 200                     |
+| `BLOCK_CHUNK_SIZE`     | Size of each chunk in bytes | 32 bytes                |
+| `num_chunks_per_share` | Chunks per share            | 125                     |
+| `total_chunks`         | Total chunks in the block   | 25,000                  |
+
+- Output: Error corrected encoded blocks, and `BlockInfo` to be submitted to the parent shard.
+
+#### 4. Block Info Submission
+
+- Input: `BlockInfo` from a child shard.
+- Any node in the shard (from farmers to full-nodes) are allowed to create the `BlockInfo`
+  transaction that submits the block information into the parent's shard history.
+- The corresponding piece of the block should be stored in the shard and made retrievable via a
+  peer-to-peer network protocol (DSN).
+- When the transaction in the parent
+- Output: Transaction in the parent shard's mempool.
+
+#### 5. Shard block proof in parent
+
+- Input: Transaction with `BlockInfo` from a child shard.
+- The execution of the `BlockInfo` transaction when included in a block of the parent shards commits
+  the block from the child shard to the history of the parent by including it on-chain into the
+  `shard_blocks[]` table.
+- To verify the `BlockInfo`, nodes in the parent shard:
+  - Ensure `block_index` is the immediate block after last committed block number in the parent
+    shard.
+- Output: `BlockInfo` from child shard included in `shard_blocks[]` table of the parent shard.
+
+#### 6. Segment Construction
+
+- Input: Buffered data of size `RAW_RECORD_SIZE`.
+- When the buffer (after encoding) contains enough data to fill a record of `RAW_RECORD_SIZE` bytes,
+  it is archived:
+  1. Split the record into `d = 2^15 =NUM_CHUNKS` chunks of size `SAFE_BYTES`.
```

Comment:
Sorry, but `SAFE_BYTES` is gone as of https://github.com/nazar-pc/abundance/pull/182

Also:
```suggestion
 1. Split the record into `NUM_CHUNKS = 2^15` chunks 32 bytes each.
```